### DISCUSSION TITLE
Need support for lpc55s16 internal temperature measurement

### DISCUSSION THREAD

**Discussion by Aravinthkalai** [zephyrproject-rtos/zephyr#59062] [Q&A] [answered]
Hello Everyone !

Greetings of the day !

I'm using lpcxpresso55s16 development board, I need to capture the internal temperature for that I searched on stack but I'm unable to find any sample code or procedure to implement, if anyone know how to read or implement  internal temperature sensor please share with me, Thanks in Advance !

Board : lpcxpresso55s16
zephyr version : 3.3.99

**Comment by DerekSnell** [comment]
Hi @Aravinthkalai ,
I am not aware of any Zephyr sample that measures the temperature with the ADC driver.  However, NXP's MCUXpresso SDK does have this [ADC temperature_measurement example](https://github.com/nxp-mcuxpresso/mcux-sdk-examples/tree/main/lpcxpresso55s16/driver_examples/lpadc/temperature_measurement).  The Zephyr HAL uses the same MCUXpresso SDK drivers.  So you can use this SDK example as a reference for how to measure the temperature with the ADC on the LPC55S16.

Best regards

**Reply by Aravinthkalai** [reply]
Hi @DerekSnell,

Thanks for your quick reply , but I can see in nordic semiconductor they implemented  [temperature measurement using zephyr](https://devzone.nordicsemi.com/f/nordic-q-a/95829/internal-temperature-measurement-with-zephyr), I tried this also but it's not working, if you have any idea please share with me.

**Comment by DerekSnell** [comment, marked as answer]
Hi @Aravinthkalai ,
Sorry, I should clarify.  The Zephyr repo does have a this [die_temp_polling sample](https://github.com/zephyrproject-rtos/zephyr/tree/main/samples/sensor/die_temp_polling).  But this requires a sensor driver that can measure the temperature using the ADC.  And I am not aware of a sensor driver available for the LPC ADC.

If you wanted to add this sensor driver, the [nxp_kinetis_temp](https://github.com/zephyrproject-rtos/zephyr/tree/main/drivers/sensor/nxp_kinetis_temp) driver is available for a similar MCU.  That could be used as a reference along with the [LPC55S16 SDK ADC temperature_measurement example](https://github.com/nxp-mcuxpresso/mcux-sdk-examples/tree/main/lpcxpresso55s16/driver_examples/lpadc/temperature_measurement) to port the sensor driver.  

Or another option is to use the ADC driver in your application to measure the temperature sensor voltage.  And then follow the code in [LPC55S16 SDK ADC temperature_measurement example](https://github.com/nxp-mcuxpresso/mcux-sdk-examples/tree/main/lpcxpresso55s16/driver_examples/lpadc/temperature_measurement) to convert that voltage to temperature.

Best regards
